src/Units/UnitState.java

```diff
@@ -138,14 +138,14 @@ public int damageHP(double damage, boolean allowOverkill)
    * When healing, rounds health up to a whole HP (e.g. 2.5 + 2 = 4.5 -> 5.0)
    * @return the change in HP
    */
-  public int alterHP(int change)
+  public int alterHP(double change)
```

Comment:
I feel like we had this whole health-vs-HP discussion already once before. HP is supposed to be non-fractional, and I'm not thrilled with the idea of making it so.
Why not just round to the nearest HP before creating the mass damage event? I get that sometimes this would negate (or accentuate) damage, but this way introduces changes to code that is exposed to other unit sets.

Reply:
That's a good point, and I agree (which is why this is a separate PR)
I really don't like the idea of rounding.

Thought: add a new function, `alterHealthPercent()` to serve this usecase?

Reply:
Wouldn't `alterHealthPercent()` just be identical to `DamageHP()` (except for possibly allowing bidirectional adjustments)?

Reply:
I'd give it an int argument (out of 100 instead of 10), and it would be non-lethal by default.

If you're not down with that, I can just close the PR, though.

Reply:
And then `MassDamageEvent` use the new call? Still not sure what benefit that would provide over `DamageHP`, since `DamageHP` already accepts a double argument.
I mean, what you are doing definitely breaks established precedent, and I'm still not sure why you need to do it to accomplish your current goals, but if I was worried about polluting the game with superfluosities, I should have stopped you before Kaiju became a thing. Just try not to upset your online fan club with collateral breaking changes I guess.

Reply:
`DamageHP` allows killing and doesn't allow healing. I don't think I want to change either of those properties, lol
 - Sidenote: I just realized `MassDamageEvent` will happily pipe healing into `damageHP()` when that will just cause an exception. So that's neat.

`AlterHP` doesn't kill, but does round up when healing

It'd be intuitive IMO for a function that takes fractional HP inputs to not round up.
The new `alterHealthPercent()` would therefore give us an endpoint to accurately model an annoying inconsistency in AW2 - Hawke's healing doesn't round fractional HP up. (None of AWBW's healing does either, heh)